src/modules/Update.rb

```diff
@@ -132,6 +132,48 @@ def ListOfRegexpsMatchesProduct(regexp_items, product)
       ret
     end
 
+    # Set vendors which are compatilbe (defined in the product description).
+    # Do not report an error if there is a vendor change while
+    # the update between these defined vendors.
+    def SetCompatibleVendors
+      return unless Mode.update # Update mode only
+
+      upgrade_settings = ProductFeatures.GetFeature("software", "upgrade")
+      # compatible vendors are not defined
+      if !upgrade_settings.is_a?(Hash) ||
+          !upgrade_settings.key?("compatible_vendors") ||
+          !upgrade_settings["compatible_vendors"] ||
+          !upgrade_settings["compatible_vendors"].is_a?(Array)
+        return
+      end
+
+      # It will not be set if the product will not be changed.
+      # e.g. SLES->SLES upgrade
+
+      # e.g. "SUSE Linux Enterprise Server 15 SP1"
+      installed_version = Installation.installedVersion["nameandversion"]
+
+      # evaluting corresponding update product
+      updated = Packages.group_products_by_status(
+        Y2Packager::Resolvable.find(kind: :product)
+      )[:updated]
+      no_product_change = updated.find do |old_product, new_product|
+        old_product.display_name == installed_version &&
+          old_product.name == new_product.name # e.g. "SLES" == "SLES"
```

Comment:
If only that was so simple...

Unfortunately some products have been renamed or changed. Some just cosmetically (like `SLE-HPC` -> `SLE_HPC`), but in some cases a module become a base product (!) (like `SLES` + `sle-module-hpc`-> `SLE_HPC`), so we cannot just compare the product names. :worried: 

See the list of the exceptions in the [Y2Packager::ProductUpgrade](https://github.com/yast/yast-yast2/blob/master/library/packages/src/lib/y2packager/product_upgrade.rb#L23-L45) class.

So actually I'm not sure if we can somehow automatically detect when the compatible vendors should be applied. Maybe in the end we will need to put the old and new product names to the `compatible_vendors` section in the `control.xml` so we know for which products this applies...